packages/transactions/tests/builder.test.ts

```diff
@@ -1061,6 +1062,118 @@ test('Estimate transaction transfer fee', async () => {
   expect(resultEstimateFee2.map(f => f.fee)).toEqual([140, 17, 125]);
 });
 
+test('Single-sig transaction byte length must include signature', async () => {
+  /*
+   * *** Context ***
+   * 1) Single-sig transaction byte length remain same due to empty message signature which allocates the space for signature
+   * 2) estimateTransactionByteLength should correctly estimate the byte length of single-sig transaction
+   */
+
+  const recipient = standardPrincipalCV('SP3FGQ8Z7JY9BWYZ5WM53E0M9NK7WHJF0691NZ159');
+  const amount = 2500000;
+  const fee = 100;
+  const nonce = 10;
+  const memo = 'test memo...';
+
+  const privateKey = 'a5c61c6ca7b3e7e55edee68566aeab22e4da26baa285c7bd10e8d2218aa3b229';
+  const publicKey = '027d28f9951ce46538951e3697c62588a87f1f1f295de4a14fdd4c780fc52cfe69';
+
+  // Create a unsigned single-sig transaction
+  const unsignedTransaction = await makeUnsignedSTXTokenTransfer({
+    recipient,
+    amount,
+    fee,
+    nonce,
+    memo: memo,
+    publicKey: publicKey,
+    anchorMode: AnchorMode.Any
+  });
```

Comment:
Does this test pass if nonce and fee are set to `null` or `undefined`? Looking at what should be the typical usage in the builders module, the only case in which a caller would need to call the estimateFee method would be when fee is undefined.

Reply:
Yes, the tests will pass.

Give it a shot:
https://github.com/hirosystems/stacks.js/blob/5be97a259eec51e582644d09f28f9b790e6621cd/packages/transactions/src/builders.ts#L553-L563

When making unsigned transaction the space for fee and nonce is allocated based on data type of fee and nonce which is IntegerType no matter what value one can set or change the fee later. 

That means transaction byte length will remains the same because it's space is allocated based on data type. No matter what value you store in it or you change anything later.

🙏

Reply:
Okay, that's fine then, thanks